linkerd/proxy/transport/src/lib.rs

```diff
@@ -27,7 +27,23 @@ fn set_nodelay_or_warn(socket: &TcpStream) {
 }
 
 fn set_keepalive_or_warn(tcp: &TcpStream, ka: Option<Duration>) {
-    if let Err(e) = tcp.set_keepalive(ka) {
+    // XXX(eliza): we now have to do "socket2 nonsense" here because tokio
+    // doesn't have `set_keepalive` any more...
```

Comment:
Replace this with a `// Safety:`? Or are we planning to get this upstreamed before merge?

Reply:
The hope was that this would be upstreamed, but it seems like that's not the case. However, it _does_ look like we will at least be able to convert tokio sockets into std sockets safely in the future.

Reply:
added a nicer comment in cb9e35c30d7671f4ecf0684c38473296193af86b